qualtran/bloqs/qft/qft_text_book.py

```diff
@@ -48,6 +70,9 @@ def decompose_from_registers(
         self, *, context: cirq.DecompositionContext, q: NDArray[cirq.Qid]  # type: ignore[type-var]
     ) -> Iterator[cirq.OP_TREE]:
         yield cirq.H(q[0])
+        if self.bitsize == 1:
+            yield cirq.H(q[0])
+            return
```

Comment:
Why do we need this ?

Reply:
good point, I was mostly following the `qft_phase_gradient` pattern and figured the explicit early return wouldnt hurt. I will remove it though since its unnecessary.